scipy/signal/signaltools.py

```diff
@@ -163,28 +162,28 @@ def correlate(in1, in2, mode='full'):
     elif not in1.ndim == in2.ndim:
         raise ValueError("in1 and in2 should have the same dimensionality")
 
-    if mode == 'valid':
-        _check_valid_mode_shapes(in1.shape, in2.shape)
-        # numpy is significantly faster for 1d
-        if in1.ndim == 1 and in2.ndim == 1:
-            return np.correlate(in1, in2, mode)
+    # numpy is significantly faster for 1d
+    if in1.ndim == 1 and in2.ndim == 1 and (in1.size >= in2.size):
```

Comment:
I guess not, since swapping can't change the way they choose the output size.

Still, this should make it work for modes other than `same`?

`if in1.ndim == in2.ndim == 1 and (in1.size >= in2.size or mode != 'same'):`